test/integration/theme-schema.test.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import Ajv from 'ajv-draft-04';
+
+/**
+ * Internal dependencies
+ */
+import themeSchema from '../../schemas/json/theme.json';
+
+describe( 'theme.json schema', () => {
+	const ajv = new Ajv( {
+		// Used for matching unknown blocks without repeating core blocks names
+		// with patternProperties in settings.blocks and settings.styles
+		allowMatchingProperties: true,
+	} );
+
+	it( 'compiles in strict mode', async () => {
+		const result = ajv.compile( themeSchema );
```

Comment:
See [point 3 from this discussion](https://github.com/WordPress/gutenberg/pull/43801#issuecomment-1236591047) about why `compile()` is being used here instead of `validate()`:

> 3. The validation code no longer uses an object to validate the schema. This ended up being trickier than expected. [Your suggested code change](https://github.com/WordPress/gutenberg/pull/43801#discussion_r961881255) using `validate` ran successfully, but did not report and strict errors on `theme.json`'s schema. For example, I tried adding invalid properties like `"test": "abc"` to the schema which should fail [Ajv's `strictSchema`](https://ajv.js.org/options.html#strictschema), but no errors were reported. The only way I could get the meta-schema to be validated was by validating an actual data structure or using `.compile()`:
>    https://github.com/WordPress/gutenberg/blob/7749202fc687d5e91206707ba430d4e13238ee5d/test/integration/theme-schema.test.js#L19
>    It's unclear from Ajv's documentation why `.validate('.../draft-04#', $themeSchema)` or `.validate($themeSchema)` weren't reporting errors. The best I could find was the [documentation for `compile()`](https://ajv.js.org/api.html#ajv-compile-schema-object-data-any-boolean-promise-any) which mentions the meta-schema:
>
>    > The schema passed to this method will be validated against meta-schema unless `validateSchema` option is false. If schema is invalid, an error will be thrown. See [options](https://ajv.js.org/api.html#options).
>
>    It seems like the meta-schema is only validated during compilation, so the code has been changed to use that instead.

Reply:
@alecgeatches I found a couple notes about this in the Ajv [docs](https://ajv.js.org/api.html#ajv-validateschema-schema-object-boolean) and [issues](https://github.com/ajv-validator/ajv/issues/1434#issuecomment-822982571).

> ### ajv.validateSchema(schema: object): boolean
> Validates schema. This method should be used to validate schemas rather than `validate` due to the inconsistency of `uri` format in JSON Schema standard.

> `validateSchema` only checks “syntax” (it validated schema as JSON against metaschema), it may still be semantically incorrect.

Looks like I was wrong about using `validate` for validating against a meta schema in the first place. And it's intentional that strict mode only applies to `compile`, so we should be using that instead of `validateSchema` to validate our schema. 🙃